DevEdu.API/DevEdu.API/Controllers/MaterialController.cs

```diff
@@ -11,7 +12,11 @@ namespace DevEdu.API.Controllers
     [Route("api/[controller]")]
     public class MaterialController : Controller
     {
-        public MaterialController() { }
+        TagRepository _tagRepository;
```

Comment:
please register this class as a scoped service in DI-container (via interface) and inject it in constructor